src/dlgTriggerEditor.cpp

```diff
@@ -780,386 +952,761 @@ void dlgTriggerEditor::slot_item_selected_search_list(QTreeWidgetItem* pItem, in
+                QStringList idStringList = vu->shortVarName(varDecendent);
+                QString idString;
+                // Take the first element - to comply with lua requirement it
+                // must begin with not a digit and not contain any spaces so is
+                // a string - and it is used "unquoted" as is to be the base
+                // of a lua table
+                if (idStringList.size() > 1) {
+                    QStringList midStrings = idStringList;
+                    idString = midStrings.takeFirst();
+                    QStringListIterator itSubString(midStrings);
+                    while (itSubString.hasNext()) {
+                        QString intermediate = itSubString.next();
+                        bool isOk = false;
+                        int numberValue = intermediate.toInt(&isOk);
+                        if ( isOk && QString::number(numberValue) == intermediate ) {
+                            // This seems to be an integer
+                            idString.append(QStringLiteral("[%1]").arg(intermediate));
+                        } else {
+                            idString.append(QStringLiteral("[\"%1\"]").arg(intermediate));
+                        }
+                    }
+                } else if (idStringList.size()) {
+                    idString = idStringList.at(0);
+                }
+
+                int startPos = 0;
+                if ((startPos = name.indexOf(s, 0, (mIsSearchCaseSensitive ? Qt::CaseSensitive : Qt::CaseInsensitive))) != -1) {
                     QStringList sl;
-                    sl << "Variable" << var2->getName() << "name";
-                    pItem2 = new QTreeWidgetItem(sl);
-                    pItem2->setFirstColumnSpanned(false);
-                    pItem2->setData(0, Qt::UserRole, vu->shortVarName(var2));
-                    treeWidget_searchResults->addTopLevelItem(pItem2);
+                    sl << tr("Variable") << idString << tr("Name");
+                    parent = new QTreeWidgetItem(sl);
+                    parent->setFirstColumnSpanned(false);
+                    // We do not (yet) worry about multiple search results in the "name"
+                    setAllSearchData(parent, name, vu->shortVarName(varDecendent), SearchResultIsName, startPos);
+                    treeWidget_searchResults->addTopLevelItem(parent);
                 }
-                if (!var2->getValue().isEmpty() && (var2->getValue().indexOf(s, 0, Qt::CaseInsensitive) != -1)) {
+
+                // The additional first test is needed to exclude the case when
+                // the search term matches on the word "function" which will
+                // appear in EVERY "value" for a lua function in the variable
+                // tree widget...
+                if (value != QLatin1String("function") && (startPos = value.indexOf(s, 0, (mIsSearchCaseSensitive ? Qt::CaseSensitive : Qt::CaseInsensitive))) != -1) {
```

Comment:
A "Whole words only" will take a rethink and a significant rewrite of each of the searches - it may be possible for the user to simulate this by surrounding the search term with spaces although entering tabs instead if those are also present in the searchable material as well can be awkward.  Similarly for regular expression based searches - it is probably doable but will need a significant further revision... 🙁